src/components/YouTube.js

```diff
@@ -0,0 +1,103 @@
+import React, {useState} from 'react'
+import styled from 'styled-components'
+
+const YouTubeStyled = styled.div`
+padding: 120px;
+display: flex;
+flex-direction: column;
+justify-content: center;
+align-items: flex-start;
+
+
+.SearchBar{
+    width: 100%;
+}
+
+.searchButton{
+    width: 100%;
+}
+
+.cardDisplay{
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+}
+
+.videoCard{
+    background-color: pink;
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    width: 300px;
+    height: 200px;
+    border: 1px solid black;
+    margin: 20px;
+}
+.title{
+    font-size: 16px;
+}
+
+.channelTitle{
+    font-size: 12px;
+}
+`
+
+function Youtube() {
+    const [searchTerm, setSearchTerm]  =  useState("")
+    const [miniCardData,setMiniCardData] = useState([])
+
+        const handleChange = event => {
+            setSearchTerm(event.target.value);
+          };
+
+    const fetchData = () => {
+        fetch(`https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=10&q=${searchTerm}&type=video&key=AIzaSyDsECTCuUQSdjsST-hja25dLugnGsjKO3M`)
+        .then(res => res.json())
+        .then(data => {
+            console.log(data)
+            setMiniCardData(data.items)
+        })
+    }
```

Comment:
You can move this code to its own file and create a custom hook, let's say `useSearch`, the signature of the hook can be something like this: `const { searchResults, loading, error } = useSearch(searchTerm);`. Having this as a hook makes the code easier to maintain and reusable, you keep the component clean and you just focus on the purpose of the component